R/utility.R

```diff
@@ -49,64 +49,144 @@ linuxWrapCommands <- function(commands = c()) {
+    for (j in 1:length(jobs$value)) {
+      id[j] <- jobs$value[[j]]$id
+      state[j] <- jobs$value[[j]]$state
+      taskCounts <-
+        rAzureBatch::getJobTaskCounts(jobId = jobs$value[[j]]$id)
+      failedTasks[j] <-
+        as.integer(taskCounts$failed)
+      totalTasks[j] <-
+        as.integer(taskCounts$active + taskCounts$running + taskCounts$completed)
+
+      completed <- as.integer(taskCounts$completed)
+
+      if (totalTasks[j] > 0) {
+        if (completed > 0) {
+          status[j] <-
+            sprintf("%s %%", ceiling(completed / totalTasks[j] * 100))
+        } else {
+          status[j] <- "No tasks were run"
         }
       }
-
-      summary <-
-        sprintf(
-          "[ id: %s, state: %s, status: %d",
-          jobs$value[[j]]$id,
-          jobs$value[[j]]$state,
-          ceiling(count / length(tasks$value) * 100)
-        )
-      print(paste0(summary,  "% ]"))
+      else {
+        status[j] <- "No tasks in the job"
+      }
     }
-    else {
-      print(
-        sprintf(
-          "[ id: %s, state: %s, status: %s ]",
-          jobs$value[[j]]$id,
-          jobs$value[[j]]$state,
-          "No tasks were run."
-        )
-      )
+  }
+
+  return (
+    data.frame(
+      Id = id,
+      State = state,
+      Status = status,
+      FailedTasks = failedTasks,
+      TotalTasks = totalTasks
+    )
+  )
+}
+
+#' Get a job for the given job id
+#'
+#' @param jobId A job id
+#' @param verbose show verbose log output
+#'
+#' @examples
+#' \dontrun{
+#' getJob("job-001", FALSE)
+#' }
+#' @export
+getJob <- function(jobId, verbose = TRUE) {
+  if (is.null(jobId)) {
+    stop("must specify the jobId parameter")
+  }
+
+  job <- rAzureBatch::getJob(jobId = jobId)
+
+  metadata <-
+    list(
+      chunkSize = 1,
+      enableCloudCombine = "TRUE",
+      packages = ""
+    )
+
+  if (!is.null(job$metadata)) {
+    for (i in 1:length(job$metadata)) {
+      metadata[[job$metadata[[i]]$name]] <- job$metadata[[i]]$value
     }
   }
+
+  if (verbose == TRUE) {
+    cat(sprintf("Job Id: %s", job$id), fill = TRUE)
+    cat("\njob metadata:", fill = TRUE)
+    cat(sprintf("\tchunkSize: %s", metadata$chunkSize),
+        fill = TRUE)
+    cat(sprintf("\tenableCloudCombine: %s", metadata$enableCloudCombine),
+        fill = TRUE)
+    cat(sprintf("\tpackages: %s", metadata$packages),
+        fill = TRUE)
+  }
+
+  taskCounts <- rAzureBatch::getJobTaskCounts(jobId = jobId)
+
+  tasks <- list(
+    active = taskCounts$active,
+    running = taskCounts$running,
+    completed = taskCounts$completed,
+    succeeded = taskCounts$succeeded,
+    failed = taskCounts$failed
+  )
+
+  if (verbose == TRUE) {
+    cat("\ntasks:", fill = TRUE)
+    cat(sprintf("\tactive: %s", taskCounts$active), fill = TRUE)
+    cat(sprintf("\trunning: %s", taskCounts$running), fill = TRUE)
+    cat(sprintf("\tcompleted: %s", taskCounts$completed), fill = TRUE)
+    cat(sprintf("\tsucceeded: %s", taskCounts$succeeded), fill = TRUE)
```

Comment:
for succeeded and failed I would prepend with a double tab since they are "sub-states" of completed.
```R
cat(sprintf("\t\tsucceeded: %s", taskCounts$succeeded), fill = TRUE)
cat(sprintf("\t\tfailed: %s", taskCounts$succeeded), fill = TRUE)
```

Reply:
fixed